crates/pathfinder/src/storage/state.rs

```diff
@@ -844,20 +763,32 @@ pub struct PageOfEvents {
 
 pub struct StarknetEventsTable {}
 impl StarknetEventsTable {
-    pub fn event_data_to_bytes(data: &[EventData]) -> Vec<u8> {
-        data.iter()
-            .flat_map(|e| (*e.0.as_be_bytes()).into_iter())
-            .collect()
+    pub fn encode_event_data_to_bytes(data: &[EventData], buffer: &mut Vec<u8>) {
+        buffer.extend(data.iter().flat_map(|e| (*e.0.as_be_bytes()).into_iter()))
     }
 
-    fn event_key_to_base64_string(key: &EventKey) -> String {
-        base64::encode(key.0.as_be_bytes())
+    fn encode_event_key_to_base64(key: &EventKey, buf: &mut String) {
+        base64::encode_config_buf(key.0.as_be_bytes(), base64::STANDARD, buf);
     }
 
-    pub fn event_keys_to_base64_strings(keys: &[EventKey]) -> String {
-        // TODO: we really should be using Iterator::intersperse() here once it's stabilized.
-        let keys: Vec<String> = keys.iter().map(Self::event_key_to_base64_string).collect();
-        keys.join(" ")
+    pub fn event_keys_to_base64_strings(keys: &[EventKey], out: &mut String) {
+        // with padding it seems 44 bytes are needed for each
```

Comment:
Could you elaborate on how this is calculated?

Reply:
nope, I don't know about the padding rules. without padding it's 43, and I think the padding will force it to be 44 always.

Reply:
Where did the number come from then?

Reply:
By refreshing up on what base64 does, googling around for worst case space usage, then looking up how long strings does it actually produce with the old configuration.

Reply:
`cargo-fuzz` cannot find me a StarkHash which would have any other base64 encoded length (using `base64::STANDARD`) than 44 bytes after few million attempts, that's not conclusive though :)

Reply:
This is no longer part, we'll handle it in a follow-up.